### PR TITLE
Update routing of means journey

### PR DESCRIPTION
**Pull request by arthurashman**
## Description of change
Update routing of means journey

Stitch together capital pages as placeholder route
Connect outgoings to capital
Add routing logic to Income as per https://miro.com/app/board/uXjVO75Jwao=/?moveToWidget=3458764570607376929&cot=14

## How to manually test the feature
Check means journey skips to evidence upload if case type is appeal to CC (no changes) aka appeal_to_crown_court after the employment page (or job lost in custody page if triggered)

Check dependants skipped based on: https://www.figma.com/file/thkvnDBbQbHlSqzt1TE3lk?node-id=7035:54438&mode=design#651514930

Check manage_no_income page is not shown if any payments/benefits are declared 
(I did consider adding a check if income < £12,475 == 'yes' too but it would take us to a cul-de-sac where income > £12,475 && no payments declared  and we wouldn't show the manage_no_income page and the user hasn't been contested on the lack of payments inconsistency)

### CODE REVIEW

File: app/forms/steps/capital/property_type_form.rb [hunk 1]
```diff
@@ -14,6 +14,7 @@ def choices
       def persist!
         # TODO: If none, go to next step
         # If type selected, build new properties form
+        true
```
Comment (arthurashman) [inline app/forms/steps/capital/property_type_form.rb:17]: Just so capital pages can be joined up

File: app/services/decisions/income_decision_tree.rb [hunk 2]
```diff
@@ -105,16 +125,44 @@ def ended_employment_within_three_months?
...
+      crime_app.case.case_type == CaseType::APPEAL_TO_CROWN_COURT.to_s
+    end
+
+    def summary_only?
+      crime_app.case.case_type == CaseType::SUMMARY_ONLY.to_s
+    end
+
     def income_below_threshold?
-      form_object.income_above_threshold.no?
+      crime_app.income.income_above_threshold == 'no'
```
Comment (arthurashman) [inline app/services/decisions/income_decision_tree.rb:137]: Do we need to account for if this answer is nil?

File: db/schema.rb
Comment (arthurashman) [inline db/schema.rb:1]: Do I need to scrap these changes? Unsure where they've come from/where we're at with any of these tables!

File: app/services/decisions/income_decision_tree.rb [hunk 3]
```diff
@@ -105,16 +125,44 @@ def ended_employment_within_three_months?
...
     end
 
     def no_property?
-      form_object.client_owns_property.no?
+      crime_app.income.client_owns_property == 'no'
+    end
+
+    def no_savings?
+      crime_app.income.has_savings == 'no'
+    end
```
Comment (moseedat) [inline app/services/decisions/income_decision_tree.rb:150]: Should this method be positive instead so that it is .savings?

File: app/services/decisions/income_decision_tree.rb [hunk 4]
```diff
@@ -105,16 +125,44 @@ def ended_employment_within_three_months?
...
+
+    def payments
+      crime_app.income_payments + crime_app.income_benefits
+    end
+
+    def crime_app
+      form_object.crime_application
+    end
```
Comment (moseedat) [inline app/services/decisions/income_decision_tree.rb:158]: Does the full name crime_application clash with another method?

File: app/services/decisions/income_decision_tree.rb [hunk 5]
```diff
@@ -105,16 +125,44 @@ def ended_employment_within_three_months?
...
+
+    def dependants_relevant?
+      if income_below_threshold? && no_frozen_assets?
+        !(summary_only? || (no_property? && no_savings?))
+      else
+        true
+      end
     end
```
Comment (moseedat) [inline app/services/decisions/income_decision_tree.rb:166]: Is there a way to simplify this?

File: app/services/decisions/income_decision_tree.rb (hunk 4 above)
Comment (arthurashman) [reply app/services/decisions/income_decision_tree.rb:158]: Nope, just trying to keep it short. I can unabbreviate if you thing better

File: app/services/decisions/capital_decision_tree.rb [hunk 6]
```diff
@@ -2,14 +2,17 @@ module Decisions
...
+      when :property_type
+        # TODO: Add next step
+        edit(:saving_type)
       when :saving_type
         after_saving_type(form_object.saving)
       when :savings
         # TODO: Add next step
-      when :property_type
-        # TODO: Add next step
+        edit(:premium_bonds)
```
Comment (timpeat) [inline app/services/decisions/capital_decision_tree.rb:12]: Be good to add an explicit TODO comment where a destination has been added in order to get the flow working.

File: app/services/decisions/capital_decision_tree.rb [hunk 7]
```diff
@@ -2,14 +2,17 @@ module Decisions
...
       when :premium_bonds
         # TODO: Add next step
+        edit('/steps/case/urn')
```
Comment (timpeat) [inline app/services/decisions/capital_decision_tree.rb:15]: Ditto

File: app/services/decisions/income_decision_tree.rb [hunk 8]
```diff
@@ -46,6 +43,8 @@ def after_employment_status
       if not_working?
         if ended_employment_within_three_months?
           edit(:lost_job_in_custody)
+        elsif appeal_no_changes?
```
Comment (timpeat) [inline app/services/decisions/income_decision_tree.rb:46]: Should this change be part of discrete change set?

File: app/services/decisions/income_decision_tree.rb (hunk 5 above)
Comment (arthurashman) [reply app/services/decisions/income_decision_tree.rb:166]: I'n not really sure how tbh! It's all of these decision diamonds https://miro.com/app/board/uXjVO75Jwao=/?moveToWidget=3458764570607377326&cot=14

File: app/services/decisions/income_decision_tree.rb (hunk 4 above)
Comment (timpeat) [reply app/services/decisions/income_decision_tree.rb:158]: I think better not use this shorter alias @arthurashman .

File: app/services/decisions/income_decision_tree.rb (hunk 2 above)
Comment (timpeat) [reply app/services/decisions/income_decision_tree.rb:137]: Why has this change been made?

File: app/services/decisions/income_decision_tree.rb [hunk 9]
```diff
@@ -105,16 +125,44 @@ def ended_employment_within_three_months?
...
+    end
+
     def income_below_threshold?
-      form_object.income_above_threshold.no?
+      crime_app.income.income_above_threshold == 'no'
     end
 
     def no_frozen_assets?
-      form_object.has_frozen_income_or_assets.no?
+      crime_app.income.has_frozen_income_or_assets == 'no'
```
Comment (timpeat) [inline app/services/decisions/income_decision_tree.rb:141]: Again, what is the purpose of this change? Do we have a type casting issue?

File: app/services/decisions/income_decision_tree.rb (hunk 3 above)
Comment (timpeat) [reply app/services/decisions/income_decision_tree.rb:150]: Again, I'm not sure what the purpose of this change is.

**Review by timpeat — CHANGES_REQUESTED** [review]

File: app/services/decisions/income_decision_tree.rb (hunk 9 above)
Comment (arthurashman) [reply app/services/decisions/income_decision_tree.rb:141]: Because originally these were helper methods just used by the forms that collected those values, so could be accessed from the form_object. I now need them on later forms, so thee data is on the crime_application.income, not on the form_object

File: app/services/decisions/income_decision_tree.rb (hunk 8 above)
Comment (arthurashman) [reply app/services/decisions/income_decision_tree.rb:46]: It can be if you want. I see all of these things as logic gates to which journey the user is taken through the means flow. Is there a reason you think it best to raise as a separate PR?

File: app/services/decisions/income_decision_tree.rb [hunk 10]
```diff
@@ -55,48 +54,69 @@ def after_employment_status
...
       else
-        edit(:manage_without_income)
+        determine_showing_no_income_page
       end
     end
 
     def edit_dependants(add_blank: false)
-      dependants = form_object.crime_application.dependants
+      dependants = crime_application.dependants
       dependants.create! if add_blank || dependants.empty?
```
Comment (timpeat) [inline app/services/decisions/income_decision_tree.rb:107]: Note: we've agreed to avoid creating records etc in the decision tree. Not part of this PR, just for future reference.

**Review by timpeat — APPROVED** [review]
Thanks @arthurashman.
